.github/workflows/sycl_nightly.yml

```diff
@@ -48,6 +54,41 @@ jobs:
       retention-days: 90
       lts_matrix: ${{ needs.test_matrix.outputs.lts_wn_matrix }}
 
+      # We use the build for the release, override its name.
+      artifact_archive_name: sycl_windows.tar.gz
+
+  nightly_release:
+    name: Nightly Release
+    if: ${{ github.ref_name == 'sycl' }}
+    needs: [ubuntu2204_build_test, windows_default]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/download-artifact@v3
+      with:
+        name: sycl_linux_default
+    - uses: actions/download-artifact@v3
+      with:
+        name: sycl_windows_default
+    - name: Compute tag
+      id: tag
+      run: |
+        if [ "${{ github.event_name == 'schedule' }}" == "true" ]; then
+          echo "TAG=$(date +'%Y-%m-%d')" >> "$GITHUB_OUTPUT"
+        else
+          # TODO: Use date of the commit?
+          echo "TAG=$(date +'%Y-%m-%d')-${GITHUB_SHA::7}" >> "$GITHUB_OUTPUT"
+        fi
```

Comment:
What is the purpose of a "nightly" release?
For regular releases, we use semantic versioning, which should work pretty well for nightly releases too.
What is the problem with using SHA for scheduled releases as well?